src/Microsoft.OData.Core/DependsOnIdsTracker.cs

```diff
@@ -0,0 +1,90 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="DependsOnIdsTracker.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData
+{
+    using System.Collections.Generic;
+    using System.Diagnostics;
+
+    /// <summary>
+    /// This class is to keep track of dependsOn ids and associated change set state,
+    /// and to return dependsOn ids pertaining to current state of the batch processing.
+    /// </summary>
+    internal sealed class DependsOnIdsTracker
+    {
+        /// <summary>
+        /// List of top-level dependsOn ids seen so far.
+        /// </summary>
+        private readonly IList<string> topLevelDepeondsOnIds;

```

Comment:
>topLevelDepeondsOnIds [](start = 39, length = 21)

Spelling; should be topLevelDependsOnIds

Reply:
oops! fixed.

---
In reply to: [159333403](https://github.com/OData/odata.net/pull/1020#discussion_r159333403) [](ancestors = 159333403)